pkg/healthcheck/healthcheck_multicluster.go

```diff
@@ -394,8 +498,42 @@ func (hc *HealthChecker) checkDaisyChains() error {
 	return &SkipError{Reason: "not checking muticluster"}
 }
 
+func (hc *HealthChecker) checkClusterGatewaysHealth() error {
+
+	var offendingGateways []string
+	options := metav1.ListOptions{
+		LabelSelector: hc.gatewayComponentsSelector(),
```

Comment:
I don't think we should be relying on this label selector since this couples the check to the specific implementation of the linkerd-gateway.  Instead, I'd consider swapping the order of this check and the next one:

First check that all exported services reference a valid gateway.  If there are 1 or more exported services then enable TargetCluster checking.  

Next, check that all the gateways referenced by the exported services are healthy.  To check health, I think it's sufficient to ensure that the gateway service has non-empty endpoints.  I don't think we need to fetch the deployment at all (in fact, we shouldn't be assuming that the gateway is a deployment, it could be e.g. a daemonset)